server/request.go

```diff
@@ -247,14 +308,14 @@ func (r *RpcRequest) sendTxToRelay() {
 
 	txTo := r.tx.To()
 	if txTo == nil {
-		r.writeRpcError("invalid target")
+		r.writeRpcError("invalid target", InternalError)
 		return
 	}
 
 	minNonce, maxNonce := r.GetAddressNonceRange(r.txFrom)
 	if r.tx.Nonce() < minNonce || r.tx.Nonce() > maxNonce+1 {
 		r.log("[sendTxToRelay] invalid nonce for %s from %s - want: [%d, %d], got: %d", txHash, r.txFrom, minNonce, maxNonce+1, r.tx.Nonce())
-		r.writeRpcError("invalid nonce")
+		r.writeRpcError("invalid nonce", InternalError)
```

Comment:
It's a user error, so should probably be `InvalidRequest` rather than `InternalError`